File: app/javascript/channels/room_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("RoomChannel", {

  // フロント側からサーバー側を監視できているかを確認できたときに動く関数
  connected() {
    // Called when the subscription is ready for use on the server
    // console.log('test')
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received(data) {
    // Called when there's incoming data on the websocket for this channel
  }
});

File: app/channels/room_channel.rb
class RoomChannel < ApplicationCable::Channel

  # サーバー側からフロントを監視できているかを確認できたときに動くメソッド
  def subscribed
    # 配信する部屋名を決定
    stream_from "room_channel"

    #5.times { puts "***test***" }
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
